src/lex.cpp

```diff
@@ -1243,8 +1248,11 @@ void lexer::skip_block_comment() {
 
 void lexer::skip_line_comment() {
   QLJS_ASSERT((this->input_[0] == '/' && this->input_[1] == '/') ||
-              (this->input_[0] == '#' && this->input_[1] == '!'));
-  for (char8* c = this->input_ + 2;; ++c) {
+              (this->input_[0] == '#' && this->input_[1] == '!') ||
+              (this->input_[0] == '<' && this->input_[1] == '!' &&
+               this->input_[2] == '-' && this->input_[3] == '-'));
+  uint8_t skip_length = this->input_[0] == '<' ? 4 : 2;
+  for (char8* c = this->input_ + skip_length;; ++c) {
```

Comment:
For future reference: I like to avoid unsigned types like `uint8_t` when doing math. The conversion rules often bite you! (Doesn't matter here, though.)
```suggestion
  int skip_length = this->input_[0] == '<' ? 4 : 2;
  for (char8* c = this->input_ + skip_length;; ++c) {
```